charts/replicated-library/templates/lib/_labels.tpl

```diff
@@ -15,40 +15,40 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
     {{- end }}
   {{- end }}
 {{- end -}}
-
 {{/* Selector labels shared across objects */}}
 {{- define "replicated-library.labels.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "replicated-library.names.appname" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end -}}
-
 {{- define "replicated-library.labels.serviceSelectorLabels" -}}
   {{- $serviceValues := . -}}
-
   {{- if hasKey . "ObjectValues" -}}
     {{- with .ObjectValues.values -}}
       {{- $serviceValues = . -}}
     {{- end -}}
   {{ end -}}
-
-  {{- if $serviceValues.appName }}
-    {{- $matchingAppFound := false -}}
-
-    {{- range $appName, $appValues := .Values.apps }}
-      {{- if and $appValues.enabled (eq $appName $serviceValues.appName) (ne $matchingAppFound true) -}}
-        {{- $matchingAppFound = true -}}
-app.kubernetes.io/name: {{ $appName }}
+  {{- if $serviceValues.selector -}}
+{{ toYaml $serviceValues.selector }}
+  {{- else -}}
+    {{- if $serviceValues.appName }}
+      {{- range $serviceValues.appName }}
+        {{- $name := . -}}
+        {{- $matchingAppFound := false -}}
+        {{- range $appName, $appValues := $.Values.apps }}
+          {{- if and $appValues.enabled (eq $appName $name) (ne $matchingAppFound true) -}}
+            {{- $matchingAppFound = true -}}
+{{ printf "app.kubernetes.io/name: %s\n" $appName }}
+          {{- end -}}
+        {{- end -}}
+        {{- if (ne $matchingAppFound true) -}}
+          {{- fail (printf "Matching app for AppName (%s) was not found" $serviceValues.appName) }}
+        {{- end -}}
+      {{- end -}}
 app.kubernetes.io/instance: {{ $.Release.Name }}
-      {{- end }}
-    {{- end }}
-
-    {{- if (ne $matchingAppFound true) -}}
-      {{- fail (printf "Matching app for AppName (%s) was not found" $serviceValues.appName) }}
-    {{- end }}
-  
-  {{- else }}
-app.kubernetes.io/name: {{- include "replicated-library.names.fullname" . -}}
+    {{- else -}}
+app.kubernetes.io/name: {{ .ObjectName }}
```

Comment:
It looks like you've changed this to ObjectName when it was fullname before. ObjectName won't include any prefix the user has specified like the release name. Was this intentional? Or maybe this use to be $appName and now it's .ObjectName?

Without testing this myself it's a little hard for me to just intuit what this part of the change does.

Reply:
Changing it back to test what happens; I now see that with your changes from yesterday I get `app.kubernetes.io/releasename-chartname-appname` but the labels applied to `app.kubernetes.io/name` on the deployment resource don't match that, being only `app.kubernetes.io/name=appname` is this the intention?

Reply:
That doesn't sound intended, I'm going to need to see specifically which labels we're talking about here. Let me try this out and see what the results are.